google-cloud-firestore/acceptance/firestore/watch_test.rb

```diff
@@ -15,6 +15,56 @@
 require "firestore_helper"
 
 describe "Watch", :firestore_acceptance do
+  it "watches a limit query" do
+    watch_col = root_col.doc("watch-limit").col("watch-query")
+
+    watch_col.doc("int 1").create(val: 1)
+    watch_col.doc("int 2").create(val: 2)
+    watch_col.doc("int 3").create(val: 3)
+
+    snps = []
+    listener = watch_col.order(:val).limit(2).listen { |snp| snps << snp }
+
+    wait_until { snps.count == 1 }
+
+    watch_col.doc("int 0").create(val: 0)
+
+    wait_until { snps.count == 2 }
+
+    listener.stop
+
+    _(snps.count).must_equal 2
+    _(snps[0].count).must_equal 2
+    _(snps[0].docs.map(&:document_id)).must_equal ["int 1", "int 2"]
+    _(snps[1].count).must_equal 2
+    _(snps[1].docs.map(&:document_id)).must_equal ["int 0", "int 1"]
+  end
+
+  it "watches a limit_to_last query" do
+    watch_col = root_col.doc("watch-limit_to_last").col("watch-query")
+
+    watch_col.doc("int 1").create(val: 1)
+    watch_col.doc("int 2").create(val: 2)
+    watch_col.doc("int 3").create(val: 3)
+
+    snps = []
+    listener = watch_col.order(:val).limit_to_last(2).listen { |snp| snps << snp }
+
+    wait_until { snps.count == 1 }
+
+    watch_col.doc("int 4").create(val: 4)
+
+    wait_until { snps.count == 2 }
+
+    listener.stop
+
+    _(snps.count).must_equal 2
+    _(snps[0].count).must_equal 2
+    _(snps[0].docs.map(&:document_id)).must_equal ["int 3", "int 2"]
```

Comment:
Can we make sure this test verifies that the order is "int 2", "int 3"?

Reply:
Thanks for catching that, will do. I'll also change line 65 below to `["int 3", "int 4"]`

Reply:
@schmidt-sebastian Can you see the mistake I made in the implementation that causes this result? This is an integration test that currently passes.

Reply:
I can take a look at the code if you need me to, but I haven't yet. My strong suspicion is that you are using the flipped query but not re-flipping the results (you always need to reverse the results if you are flipping the orderBy constraints on the user's behalf). The way I do this is Node and Java is that I use the pre-flipped query to process the local results.

Reply:
That sounds likely, thank you.

Reply:
@schmidt-sebastian I updated `Inventory` to reverse the results for Query#limit_to_last queries. I'm sorry I missed that the first time. Please take a look at the additional assertions for `changes` ordering to verify that it is correct. (I'm not sure if `changes` ordering should reflect query `orderBy`.)

Reply:
This looks correct now. It is slightly different than what I have implemented since the Watch client in Ruby uses the reversed query Proto for its ordering, rather than the user-specified Query order. In the end this does not matter though, since we only need to match the user's order by clause. With the reversal this is now working as intended.

Reply:
@schmidt-sebastian Thank you for your help on this PR! I'm going to add samples for this feature (as in GoogleCloudPlatform/python-docs-samples#4196), then ask @dazuma to review for Ruby concerns.

Reply:
Here are the 3 samples PRs I found:

Go PR: GoogleCloudPlatform/golang-samples#1599
Python PR: GoogleCloudPlatform/python-docs-samples#4196
Node PR: googleapis/nodejs-firestore#1113

Reply:
Added sample in 13c2253